lib/repetitive-task.js

```diff
@@ -52,20 +53,26 @@ class RepetitiveTask {
   }
 
   __process () {
+    let processStartTime
     this.isProcessing = this._getTask()
       .then((task) => {
         if (!task) {
           this.logger.info('No pending tasks')
           return
         }
-
+        this.currentTask = task
         this.logger.info(this._getLogData(task), 'processing task')
+        processStartTime = process.hrtime()
         return this._process(task)
       })
       .catch((error) => {
         this.logger.error({ stack: error.stack }, 'Error processing task')
       })
       .then(() => {
+        const processElapsedTime = { elapsed_time: hrtimeToMs(process.hrtime(processStartTime)) }
+        const logInfo = Object.assign(this._getLogData(this.currentTask), processElapsedTime)
```

Comment:
If you want to guarantee that you can relate it back, you need to generate a unique ID for the task here and log it both times. The task itself can have no data or tons of data. In the no data case, you can't relate it back and in the lots of data case, you're logging a lot of information for no reason.